build.rs

```diff
@@ -63,7 +167,20 @@ fn extract_tar_gz(path: &Path, output_dir: &Path) {
 
 /// Determines whether the CRT is being statically or dynamically linked.
 fn static_crt() -> bool {
-    var("CARGO_ENCODED_RUSTFLAGS")
-        .unwrap_or_default()
-        .contains("target-feature=+crt-static")
-}
\ No newline at end of file
+    env::var("CARGO_ENCODED_RUSTFLAGS")
+        .map(|flags| flags.contains("target-feature=+crt-static"))
+        .unwrap_or(false)
+}
+
+/// Get global cache path for downloaded file.
+fn get_cached_path() -> PathBuf {
+    const CACHE_FOLDER_NAME: &str = "mach_dxcompiler_rs";
+    if let Ok(cache_root) = env::var("CARGO_TARGET_DIR") {
```

Comment:
Has this been tested to make sure that the `Ok` branch executes? I ask because I don't think Cargo passes `CARGO_TARGET_DIR` to build scripts; it is an input to `cargo build` only: https://github.com/rust-lang/cargo/issues/9661

Reply:
> Has this been tested to make sure that the `Ok` branch executes? I ask because I don't think Cargo passes `CARGO_TARGET_DIR` to build scripts; it is an input to `cargo build` only: [rust-lang/cargo#9661](https://github.com/rust-lang/cargo/issues/9661)

No, it doesn't executes, I added this to keep the same directory, and set up the target manually is not appropriate here I guess.

> Considering the confusion around this, I am left wondering whether caching is really worth it. For consumers of `mach-dxcompiler-rs`, the download will only happen once (when the user first adds the crate or if `cargo clean` is run). Decoupling when the build script runs from when the file is downloaded might cause confusion.
> 
> For developing `mach-dxcompiler-rs` ourselves, well, this is a small crate. Is build time for this crate when developing locally really going to be an issue? What other features would we even want to add to this crate?

I thought global cache is a good thing only if there are multiple projects that use this crate, and after your considerarions I think we only need to keep it in `OUT_DIR`.